### PR TITLE
Add test for Jupyter notebooks

### PR DESCRIPTION
**Pull request by snorfalorpagus**
This PR adds a basic test for drawing a network diagram in d3 from Jupyter.

Also fixes a regression introduced in the notebook module (related to `model.node`).

This adds quite a few dependencies to the pywr package - is this something we're OK with? I don't see any need to install jupyter on the cluster...

Build is failing on appveyor for no reason...

### CODE REVIEW

**Comment by jetuk** [comment]
This is ready for review and merge despite the appveyor failure?

**Comment by snorfalorpagus** [comment]
I believe so. I've checked - the extra dependencies only add a minute on to the build time. In the future we may have more detailed tests that need Jupyter.

File: conda-recipe/bld.bat [hunk 1]
```diff
@@ -2,5 +2,5 @@ set LIB=%LIBRARY_LIB%;.\lib;%LIB%
 set LIBPATH=%LIBRARY_LIB%;.\lib;%LIBPATH%
 set INCLUDE=%LIBRARY_INC%;%INCLUDE%
 
-"%PYTHON%" setup.py build_ext -I"%LIBRARY_INC%" -L"%LIBRARY_LIB%" install --with-glpk --with-lpsolve
+"%PYTHON%" setup.py build_ext -I"%LIBRARY_INC%" -L"%LIBRARY_LIB%" install --with-glpk --with-lpsolve --single-version-externally-managed --record=record.txt
```
Comment (jetuk) [inline conda-recipe/bld.bat]: `record.txt` isn't an external that needs to be provided?

Comment (snorfalorpagus) [inline conda-recipe/bld.bat]: No, it's an output (that isn't used). I just copy+pasted from a conda-forge build. It's something to do with conda's handling off eggs I think.

**Pull request merged**